data/src/emulator.js

```diff
@@ -636,11 +755,41 @@ class EmulatorJS {
                     this.elements.bottomBar.loadSavFiles[0].style.display = "none";
                 }
 
+                // The core decompression is now handled by checkCompression which already caches the result
+                // No need for additional core-specific caching - this would create duplicates
+                console.log(`[EJS Core] Core decompression complete (cached by checkCompression)`);
+
                 this.initGameCore(js, wasm, thread);
             });
         }
+
+        // Helper function to generate cache key for core data
+        const generateCoreDataCacheKey = (data) => {
```

Comment:
This is the third time this one function is in this file, I think we should make a utils file of some sort.